### PR TITLE
add custom text editor

### PR DESCRIPTION
**Pull request by hydra0210**
### What's new

This creates a custom editor that can mask it's input when set. It can also be set to only allow numeric characters by calling  `editor.IsNumeric()` function

### CODE REVIEW

File: ui/themes/materialplus/buffer.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	e.moveGap(len(s))
+	copy(e.text[e.caret:], s)
+	e.gapstart += len(s)
+	e.changed = e.changed || len(s) > 0
+	e.dump()
+}
+
+func (e *editBuffer) dump() {
+	if bufferDebug {
+		fmt.Printf("len(e.text) %d e.len() %d e.gapstart %d e.gapend %d e.caret %d txt:\n'%+x'<-%d->'%+x'\n", len(e.text), e.len(), e.gapstart, e.gapend, e.caret, e.text[:e.gapstart], e.gapLen(), e.text[e.gapend:])
```
Comment (oluwandabira) [inline ui/themes/materialplus/buffer.go:150]: Use the `log`

File: ui/themes/materialplus/editor.go [hunk 2]
```diff
@@ -0,0 +1,868 @@
...
+	"gioui.org/op/paint"
+	"gioui.org/text"
+	"gioui.org/unit"
+	"gioui.org/widget"
+
+	"golang.org/x/image/math/fixed"
+)
+
+// Editor implements an editable and scrollable text area.
+type Editor struct {
```
Comment (oluwandabira) [inline ui/themes/materialplus/editor.go]: Is gioui's Editor not editable and scroll-able?

Comment (yaowork) [reply ui/themes/materialplus/editor.go]: this is a modification of the gioui's Editor codes to add the functionality of masking characters for use as a password editor. Gio doesn't provide this functionality out of the box

Comment (oluwandabira) [reply ui/themes/materialplus/editor.go]: Then the comment should reflect that. And if it is for passwords, why does it need to be scrollable?

**Comment by oshorefueled** [comment]
password masking feature to be pushed upstream

**Pull request closed without merging**